SpatialGDK/Source/SpatialGDK/Public/Schema/AuthorityIntent.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/Component.h"
+#include "Utils/SchemaUtils.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+
+namespace SpatialGDK
+{
+
+// The AuthorityIntent component is a piece of the Zoning solution for the UnrealGDK. For each
+// entity in SpatialOS, Unreal will use the AuthorityIntent to indicate which Unreal server worker
+// should be authoritative for the entity. No Unreal worker should write to an entity if the
+// VirtualWorkerId set here doesn't match the worker's Id. 
+struct AuthorityIntent : Component
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID;
+
+	AuthorityIntent() = default;
+
+	AuthorityIntent(uint32 InVirtualWorkerId)
+		: VirtualWorkerId(InVirtualWorkerId) {}
+
+	AuthorityIntent(const Worker_ComponentData& Data)
+	{
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.schema_type);
+
+		VirtualWorkerId = Schema_GetUint32(ComponentObject, 1);
```

Comment:
could also add a constant for the field id here in SpatialConstants

Reply:
and for the other uses of the field id below

Reply:
Done.